Add volmgr support for python-agent

yasker opened the pull request:
Still working on test.

ibuildthecloud commented on cattle/plugins/docker/storage.py:
```diff
@@ -158,6 +158,49 @@ def parse_repo_tag(image_uuid):
+        snapshot_uuid = volmgr.create_snapshot(volume_uuid)
+        snapshot.data.fields['snapshotUUID'] = snapshot_uuid
+        log.info("New snapshot for volume, uuid is %s" % snapshot_uuid)
+        volmgr.backup_snapshot(snapshot_uuid, volume_uuid)
```
Move to separate event

yasker commented:
Addressed the comments, separate snapshot handling from docker storage pool, also separate snapshot taking process with backup process.

Still working on tests. As well as idempotent checks.

yasker commented:
Check https://github.com/rancherio/python-agent/pull/108 for the latest PR
